src/tests/test_integration_tests.py

```diff
@@ -19,18 +22,35 @@ def mock_init(self): pass
         yield client
 
 def test_index_page(monkeypatch, client):  
-    def mock_get_items(self):
-        return [
+    monkeypatch.setattr(
+        'board_repository.BoardRepository.get_items', 
+        lambda self: [
             ToDoItem('123', 'Wash the car', 'To Do', datetime.now()),
             ToDoItem('456', 'Fly to space', 'Completed', datetime.now())
         ]
+    )
 
-    monkeypatch.setattr('board_repository.BoardRepository.get_items', mock_get_items)
+    mock_user(monkeypatch, client, roles=[AuthProvider.READER_ROLE])
 
     response = client.get('/')
 
     assert response.status_code == 200
 
     response_body = response.get_data(True)
     assert 'Wash the car' in response_body
-    assert 'Fly to space' in response_body
\ No newline at end of file
+    assert 'Fly to space' in response_body
+
+def test_index_page_returns_403_if_user_not_in_reader_role(monkeypatch, client):
+    mock_user(monkeypatch, client, roles=[AuthProvider.WRITER_ROLE])
+
+    response = client.get('/')
+
+    assert response.status_code == 403
+
+def mock_user(monkeypatch, client, roles):
+    user = AppUser('someUserId', roles)
+
+    monkeypatch.setattr('auth_provider.AuthProvider.get_authenticated_user', lambda self, auth_code: user)
+    monkeypatch.setattr('auth_provider.AuthProvider.get_user', lambda self, user_id: user)
+
+    client.get('/login/callback?code=someUserId')
```

Comment:
It's not obvious at first glance why this line is necessary - this might be a good candidate for a simple comment on what this is achieving!